pkg/reconciler/pipelinerun/resources/pipelinerunresolution.go

```diff
@@ -219,6 +219,11 @@ func (t *ResolvedPipelineRunTask) parentTasksSkip(facts *PipelineRunFacts) bool
 	return false
 }
 
+// IsFinalTask returns true if a task is a finally task
+func (t *ResolvedPipelineRunTask) IsFinalTask(facts *PipelineRunFacts) bool {
```

Comment:
We could make `facts.isFinalTask` public instead of wrapping it in a new function on the RPRT type.  wdyt?

Reply:
(I'm assuming this isn't prevented for some other reason reason I'm missing?)

Reply:
It just seemed to me more natural to put it that way because I was expecting to find the function closed to ```IsFinallySkipped```. I like the idea of collocating those receiver function as they seem related. But i'm not against changing the scope of ```facts.isFinalTask```.

Reply:
Nope, there is no specific reason preventing it from going public i.e `facts.isFinalTask` can be made public.

Reply:
>  I like the idea of collocating those receiver function as they seem related.

I see this a bit differently - `IsFinallySkipped` is answering a question about the whole pipeline graph which requires the graph be fully "resolved", so makes sense on the `ResolvedPipelineRunTasks` type.  `IsFinalTask` answers a question about a "fact" of an individual task of the pipeline, so makes sense to me to live on the `PipelineRunFacts` type.

Given that these two types (RPRT and PipelineRunFacts) are used so closely together in our code I don't feel super strongly either way.  I just got a bit confused when I first read it because the two funcs are named so similarly and use the same objects 😅